src/web/src/app/space/service.ts

```diff
@@ -363,73 +190,35 @@ export class SpaceService {
 		}
 	}
 
-	join(spaceId: string): Observable<boolean> {
-		let space = this.spaces.find(s => s.id === spaceId);
-		if (space) {
-			return of(space.requestsAllowed)
-				.pipe(
-					delay(2000),
-					src => this.requestWatcher.WatchFor(src)
-				)
-		} else {
-			return of(false)
-				.pipe(
-					delay(2000),
-					src => this.requestWatcher.WatchFor(src)
-				)
-		}
-	}
-
-	joinByLink(token: string): Observable<boolean> {
-		let link = this.links.find(l => l.id === token);
-		let now = new Date();
-		if (link.expiredAt > now) {
-			let space = this.spaces.find(s => s.id === link.spaceId);
-			return of(true)
-				.pipe(
-					delay(2000),
-					src => this.requestWatcher.WatchFor(src)
-				)
-		}
+	join(spaceId: string, userId: string): Observable<any> {
+		return this.http.post(this._apiUrl + "/invitation", {
+			id: 0,
+			spaceId: spaceId,
+			creatorId: userId,
+			userId: userId,
+			role: "guest",
```

Comment:
Нужно в /app/space/model.ts описать enum с ролями и использовать его значения везде в space вместо строковых значений.
```suggestion
			role: SpaceRole.Guest,
```

Reply:
Готово. Но только в одном месте, вроде, пригодилось, если я ничего не напутал.